game-state-service

```
@@ -0,0 +1,123 @@
+        self._monitor_restart_id = None
+        self._monitor_start()
+        self._state_from_file()
+
+    def _get_state_file(self):
+        path = GLib.build_filenamev([GLib.get_user_data_dir(), 'state.json'])
+        return Gio.File.new_for_path(path)
+
+    def _monitor_start(self):
+        state_file = self._get_state_file()
+        self._monitor = state_file.monitor_file(
+            Gio.FileMonitorFlags.NONE,
+            None)
+        self._monitor.set_rate_limit(self._MONITOR_RATE)
+        self._monitor_changed_id = self._monitor.connect(
+            'changed',
+            self._monitor_change_cb)
+
+    def _monitor_change_cb(self, monitor, child, other_file, event):
+        if event == Gio.FileMonitorEvent.CHANGES_DONE_HINT:
+            self._state_from_file()
+
+    def _monitor_restart_cb(self):
+        self._monitor_restart_id = None
+        self._monitor.handler_unblock(self._monitor_changed_id)
+        return False
+
+    def _state_from_file(self):
+        state_file = self._get_state_file()
+        try:
+            success, state_data, tag = state_file.load_contents(None)
+            self._state = json.loads(state_data.decode())
+        except:
+            pass
+
+    def _state_to_file(self):
+        state_data = json.dumps(self._state,
+                                sort_keys=True,
+                                indent=2,
+                                separators=(',', ': '))
+        state_file = self._get_state_file()
+
+        # always block the monitor when writing to the state file
+        if self._monitor_restart_id is None:
+            self._monitor.handler_block(self._monitor_changed_id)
+        else:
+            GLib.source_remove(self._monitor_restart_id)
+
+        try:
+            state_file.replace_contents(
+                bytes(state_data.encode()),
+                None,
+                False,
+                0,
+                None)
+        except:
```

I think you can just delete this `except: pass`